src/seedu/addressbook/commands/ListCommand.java

```diff
@@ -20,6 +24,20 @@
     @Override
     public CommandResult execute() {
         List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
-        return new CommandResult(getMessageForPersonListShownSummary(allPersons), allPersons);
+        List<ReadOnlyPerson> listOfSortedPersons = new ArrayList<>();
+        for (int i = 0; i < allPersons.size(); i++) { // Copy List from unsorted array to array that will be sorted
+            listOfSortedPersons.add(allPersons.get(i));
+        }
+        
+        for (int i = 0; i < listOfSortedPersons.size()-1; i++) // Two for loops will determine the correct position to place an entry in
+            for (int j = 0; j < listOfSortedPersons.size()-i-1; j++)
+                if (listOfSortedPersons.get(j).getName().toString().toLowerCase().compareTo(listOfSortedPersons.get(j+1).getName()
+                        .toString().toLowerCase()) > 0) // the names are converted to string and to lowercase to take advantage of the compareTo function
```

Comment:
There are too many comments here, that usually means you think your code is not clear. Thus, you need to update the code, e.g. apply abstraction. Moreover, it's better to remove magic numbers.